### DISCUSSION TITLE
Difference between ImageWMS and ImageSource

### DISCUSSION THREAD

**Discussion by alexerisov** [openlayers/openlayers#15751] [General]
Hello. Can anyone describe difference between `ImageWMS` and `ImageSource`? Both can be used as source of `ImageLayer`, but idk which one i have to choose.

**Comment by ahocevar** [comment]
The `Layer` is what you're going to display (in this case a single image for the whole map viewport), the `Source` is responsible for providing and requesting the data for that single image.

`ImageWMS` is a convenience class for requesting images from a [WMS](https://www.ogc.org/standard/wms/) service. You can always use `ImageSource`, then you can choose from service specific `createLoader()` functions (like `ol/source/wms.js`), or create your own `loader`.

**Reply by alexerisov** [reply]
Thank you 👍🏼
